www/FileWriter.js

```diff
@@ -107,8 +108,29 @@ FileWriter.prototype.write = function(data) {
         var fileReader = new FileReader();
         fileReader.onload = function() {
             // Call this method again, with the arraybuffer as argument
-            FileWriter.prototype.write.call(that, this.result);
+            FileWriter.prototype.write.call(that, this.result, true /* isPendingBlobReadResult */);
         };
+        fileReader.onerror = function (evt) {
+            // DONE state
+            that.readyState = FileWriter.DONE;
```

Comment:
Do you need to check if this FileWriter is already in the DONE (cancelled) state, as in the other callbacks below?